### PR TITLE
Integrate MAM SDK with ADAL Test App

### PR DESCRIPTION
**Pull request by jasoncoolmax**
To serve the purpose of testing True MAM scenario.

The MAM SDK is not included yet. We will explore a better solution to include large file to the repo.

For now, testing will require:
1. move the MAM SDK framework to the test app build target as embedded binary;
2. define the preprocessor macro AD_MAM_SDK_TESTING to activate the MAM SDK related code

### CODE REVIEW

File: Samples/MyTestiOSApp/MyTestiOSApp/ADTestAppAcquireTokenViewController.m [hunk 1]
```diff
@@ -28,7 +28,15 @@
 #import "ADTestAppProfileViewController.h"
 #import "ADTestAppClaimsPickerController.h"
 
-@interface ADTestAppAcquireTokenViewController () <UITextFieldDelegate>
+#if AD_MAM_SDK_TESTING
+#import <IntuneMAMWalledGarden/IntuneMAM.h>
```
Comment (midella) [inline Samples/MyTestiOSApp/MyTestiOSApp/ADTestAppAcquireTokenViewController.m]: Public release will be <IntuneMAM/IntuneMAM.h>

Comment (jasoncoolmax) [reply Samples/MyTestiOSApp/MyTestiOSApp/ADTestAppAcquireTokenViewController.m]: Yep, will change that once I get the public MAM SDK with true mam changes from you :)

File: Samples/MyTestiOSApp/MyTestiOSApp/ADTestAppAcquireTokenViewController.m [hunk 2]
```diff
@@ -28,7 +28,15 @@
...
+#endif
+
+@interface ADTestAppAcquireTokenViewController () <UITextFieldDelegate
```
Comment (oldalton) [inline Samples/MyTestiOSApp/MyTestiOSApp/ADTestAppAcquireTokenViewController.m]: nit: this looks a bit awkward, can we have whole protocol definite in if and else

File: Samples/MyTestiOSApp/MyTestiOSApp/ADTestAppAcquireTokenViewController.m [hunk 3]
```diff
@@ -28,7 +28,15 @@
...
+#if AD_MAM_SDK_TESTING
```
Comment (oldalton) [inline Samples/MyTestiOSApp/MyTestiOSApp/ADTestAppAcquireTokenViewController.m]: nit: I prefer to have ifdef, but like that is also fine

File: Samples/MyTestiOSApp/MyTestiOSApp/ADTestAppAcquireTokenViewController.m [hunk 4]
```diff
@@ -72,6 +80,10 @@ - (id)init
     [self setTabBarItem:tabBarItem];
     
     [self setEdgesForExtendedLayout:UIRectEdgeTop];
+#if AD_MAM_SDK_TESTING
+    [[IntuneMAMComplianceManager instance] setDelegate:self];
```
Comment (oldalton) [inline Samples/MyTestiOSApp/MyTestiOSApp/ADTestAppAcquireTokenViewController.m]: does MAM SDK holds a weak reference to its delegate?

File: Samples/MyTestiOSApp/MyTestiOSApp/ADTestAppAcquireTokenViewController.m (hunk 2 above)
Comment (jasoncoolmax) [reply Samples/MyTestiOSApp/MyTestiOSApp/ADTestAppAcquireTokenViewController.m]: done

File: Samples/MyTestiOSApp/MyTestiOSApp/ADTestAppAcquireTokenViewController.m (hunk 3 above)
Comment (jasoncoolmax) [reply Samples/MyTestiOSApp/MyTestiOSApp/ADTestAppAcquireTokenViewController.m]: done

File: Samples/MyTestiOSApp/MyTestiOSApp/ADTestAppAcquireTokenViewController.m (hunk 4 above)
Comment (jasoncoolmax) [reply Samples/MyTestiOSApp/MyTestiOSApp/ADTestAppAcquireTokenViewController.m]: yep

**Pull request merged**